src/App.js

```diff
@@ -1,24 +1,61 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import 'bulma/css/bulma.css';
+import foodsArr from './foods.json';
+import FoodBox from './components/FoodBox';
+import { useState } from "react"; 
+import FoodForm from './components/FoodForm';
+import Search from './components/Search';
 
 function App() {
+
+  const [foods, setFoods] = useState(foodsArr);
+  /* console.log(foods[0].name) */
+
+  const [showForm, setShowForm] = useState(false);
+
+  const createNewFood = (newFood) => {
+    setFoods((prevFood)=>{
+      return [newFood, ...prevFood];
+    });
+  }
+
+  const searchFood = (search) => {
+    const searchResults = foods.filter((food) => food.name.includes(search))
+    console.log(searchResults)
+    setFoods(searchResults);
+  }
```

Comment:
Great solution on search functionality! You almost got it!

We only need to make a little tweak 
- add another state , to hold filteredArray and set initial value to original array
```
  const [filteredFoods, setFilteredFoods] = useState(foods);
```

- In searchFood function, instead of updating food , we want to update the value of filtered array 
```
 const searchFood = (search) => {
    const searchResults = foods.filter((food) => food.name.includes(search))
    console.log(searchResults)
    setFilteredFoods(searchResults);
  }
```

Now what's the reason for having additional state... We want to always be displaying 'filtered array'.
- initially, it's value is the same original array from json / or database
- when you filter once (have letters in search input field), we filter teh ORIGINAL array, and set new value of filtered array.
- the second time you add letters to search, or remove letters -> we AGAIN filter from the ORIGINAL array. Because if were filtering from the filtered array furthermore (that's what you have at the moment, we are missign a lot of foods right?)


One last note, when you filter -> we need to make sure it's case insecitive - better user experience.
```
    const searchResults = foods.filter((food) => food.name.toLowerCase().includes(search.toLowerCase()))
```